r/tests/testthat/test-dplyr-arrange.R

```diff
@@ -0,0 +1,217 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+library(dplyr)
+
+# randomize order of rows in test data
+tbl <- slice_sample(example_data_for_sorting, prop = 1L)
+
+# use the C locale for string collation in R (ARROW-12046)
+Sys.setlocale("LC_COLLATE", "C")
```

Comment:
I would just set this globally in helper-arrow.R and don't bother unsetting it at the end. Later we can use `with_locale` to test local changes to it.

Reply:
Done in dd2e0efd05ff46fae184ebb18965a39cdbfcce96